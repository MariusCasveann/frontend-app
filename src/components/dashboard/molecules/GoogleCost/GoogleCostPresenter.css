.google-cost-card {
    display: flex;
    display: -ms-flexbox;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
}

.google-cost-card .numbers-container {
    padding-left: 20px;
    padding-top: 20px;
    display: grid;
    grid-auto-flow: column;
}

.google-cost-card .numbers-container .ant-typography {
    margin-bottom: 0.1em;
}

.google-cost-card .chart {
    padding: 15px 15px 15px;
    flex: 1;
    display: block;
    min-height: 250px;
    min-width: 320px;
}

/* Style for mobile phone */
@media only screen and (max-width: 576px) {
    .google-cost-card {
        flex-direction: column;
    }

    .google-cost-card .chart {
        max-width: 320px;
    }

    .google-cost-card .chart {
        padding: 5px 5px 5px;
    }
}
