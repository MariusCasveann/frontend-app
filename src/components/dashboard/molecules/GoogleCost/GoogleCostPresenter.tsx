import React, { useState } from 'react';
import { Card } from '../../../common/molecules/CardLayout/CardLayout';
import GoogleCostWidget from './GoogleCostWidget';
import { DeploymentStageEnum } from '../../../../model/__generated__/globalTypes';
import './GoogleCostPresenter.css';
import { FiltersContext } from '../../atoms/Filters/FiltersContext';

export default () => {
    const [maxValue, setMaxValue] = useState<number>(1000);

    return (
        <FiltersContext.Consumer>
            {({ fromDate, toDate, financialYear }) => (
                <Card title="Google costs" width={100}>
                    <div className="google-cost-card">
                        <div className="chart">
                            <GoogleCostWidget
                                stage={DeploymentStageEnum.DEV}
                                maxValue={maxValue}
                                fromDate={fromDate}
                                toDate={toDate}
                                setMaxValue={setMaxValue}
                                fiscalYear={financialYear}
                            />
                        </div>
                        <div className="chart">
                            <GoogleCostWidget
                                stage={DeploymentStageEnum.PROD}
                                maxValue={maxValue}
                                fromDate={fromDate}
                                toDate={toDate}
                                setMaxValue={setMaxValue}
                                fiscalYear={financialYear}
                            />
                        </div>
                    </div>
                </Card>
            )}
        </FiltersContext.Consumer>
    );
};
