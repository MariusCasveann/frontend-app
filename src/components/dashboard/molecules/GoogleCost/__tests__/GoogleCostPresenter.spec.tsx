import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import GoogleCostPresenter from '../GoogleCostPresenter';

jest.doMock('../../../atoms/Filters/FiltersContext', () => ({
    FiltersContext: 'FiltersContext',
    'FiltersContext.Consumer': 'FiltersContext.Consumer'
}));

// tslint:disable-next-line:no-any
const _ShallowRenderer: any = ShallowRenderer;
const renderer = new _ShallowRenderer();

const tree = renderer.render(<GoogleCostPresenter />);
const result = renderer.getRenderOutput();

describe('GoogleCostPresenter', () => {
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});
