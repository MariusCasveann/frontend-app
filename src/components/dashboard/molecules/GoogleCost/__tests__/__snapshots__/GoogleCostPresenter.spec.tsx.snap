// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleCostPresenter renders correctly 1`] = `
<Context.Consumer>
  [Function]
</Context.Consumer>
`;
